core/src/server/rpc_module.rs

```diff
@@ -266,6 +279,23 @@ impl Methods {
 		Ok(())
 	}
 
+	/// Merge two [`Methods`]'s by adding all [`MethodCallback`]s from `other`
+	/// into `self`, removing and returning any existing methods with the same
+	/// name.
+	pub fn merge_replacing(&mut self, other: impl Into<Methods>) -> Vec<(&'static str, MethodCallback)> {
```

Comment:
As discussed in the predecessor PR, this API isn't great because it's hard to know which callback that was replaced.

For instance if one has several RpcModule's and one would want to merge it into a single RpcModule that actual merge order matters and you may end up with unexpected callback is overwritten.

```rust
fn default_setup () -> RpcModule {
   let apis = Vec<RpcModule>;
   let mut api = RpcModule::new(());
   for a in apis {
      api.merge_replace(a);
   }
}

```

I would personally prefer merge_if but I guess we could have both

Reply:
unlike the previous PR, this returns an explicit list of methods overridden, which is the rust idiom for maps like this

merge_replace_if is also included now :)

Reply:
cool, sorry missed that

Reply:
no worries, i think i pushed it after you started the review